controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,428 @@
+		// The DisruptionSchedule is being deleted.
+		return ctrl.Result{}, nil
+	}
+
+	targetResourceExists, instanceDeleted, err := r.updateTargetResourcePreviouslyMissing(ctx, instance)
+	if err != nil {
+		// Error occurred during status update or deletion, requeue
+		r.log.Errorw("failed to handle target resource status", "err", err)
+		return ctrl.Result{}, err
+	}
+
+	if instanceDeleted {
+		// The instance has been deleted, reconciliation can be skipped
+		return ctrl.Result{}, nil
+	}
+
+	disruptions, err := r.getChildDisruptions(ctx, instance)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.updateLastScheduleTime(ctx, instance, disruptions); err != nil {
+		r.log.Errorw("unable to update LastScheduleTime of DisruptionSchedule status", "err", err)
+		return ctrl.Result{}, err
+	}
+
+	// Get the next scheduled run, or the time of the unproccessed run
+	// If there are multiple unmet start times, only start last one
+	missedRun, nextRun, err := r.getNextSchedule(instance, time.Now())
+	if err != nil {
+		r.log.Errorw("unable to figure out disruption schedule", "err", err)
+		// we don't really care about requeuing until we get an update that
+		// fixes the schedule, so don't return an error
+		return ctrl.Result{}, nil
+	}
+
+	// Calculate the requeue time
+	scheduledResult := ctrl.Result{RequeueAfter: time.Until(nextRun)}
+	requeueTime := scheduledResult.RequeueAfter.Round(time.Second)
+
+	r.log.Infow("calculated next scheduled run", "nextRun", nextRun.Format(time.UnixDate), "now", time.Now().Format(time.UnixDate))
+
+	if missedRun.IsZero() {
+		r.log.Infow(fmt.Sprintf("no missed runs detected, scheduling next check in %s", requeueTime))
+		return scheduledResult, nil
+	}
+
+	if !targetResourceExists {
+		r.log.Infow(fmt.Sprintf("target resource is missing, scheduling next check in %s", requeueTime))
+		return scheduledResult, nil
+	}
+
+	if len(disruptions.Items) > 0 {
+		r.log.Infow(fmt.Sprintf("cannot start a new disruption as a prior one is still running, scheduling next check in %s", requeueTime), "numActiveDisruptions", len(disruptions.Items))
+		return scheduledResult, nil
+	}
+
+	r.log.Infow("processing current run", "currentRun", missedRun.Format(time.UnixDate))
+
+	tooLate := false
+	if instance.Spec.StartingDeadlineSeconds != nil {
+		tooLate = missedRun.Add(time.Duration(*instance.Spec.StartingDeadlineSeconds) * time.Second).Before(time.Now())
+	}
+
+	if tooLate {
+		r.log.Infow(fmt.Sprintf("missed schedule to start a disruption at %s, scheduling next check in %s", missedRun, requeueTime))
+		return scheduledResult, nil
+	}
+
+	disruptionCreated, err := r.createDisruption(ctx, instance, missedRun)
+	if err != nil {
+		r.log.Warnw(fmt.Sprintf("failed to create a disruption, scheduling next check in %s", requeueTime), "err", err)
+		return scheduledResult, err
+	}
+
+	if disruptionCreated {
+		r.log.Infow("created Disruption")
+
+		instance.Status.LastScheduleTime = &metav1.Time{Time: missedRun}
```

Comment:
Why are we setting this to missedRun, instead of the disruption we just created? (there will be a gap in those times, no?)